src/client/common/process/condaService.ts

```diff
@@ -0,0 +1,89 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+import { inject, injectable, named } from 'inversify';
+import { SemVer } from 'semver';
+import { Memento } from 'vscode';
+import { IPythonApiProvider } from '../../api/types';
+import { IFileSystem } from '../platform/types';
+import { GLOBAL_MEMENTO, IMemento } from '../types';
+import { createDeferredFromPromise } from '../utils/async';
+
+const CACHEKEY_FOR_CONDA_INFO = 'CONDA_INFORMATION_CACHE';
+
+@injectable()
+export class CondaService {
```

Comment:
Exposes the CondaVersion, CondaFile behind a class with caching & wraps the Python API call (possible its not available, etc)